tests/console/verify_secure_boot.pm

```diff
@@ -0,0 +1,31 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Verify that secure boot is set as expected.
+# Maintainer: Sofia Syrianidou <ssyrianidou@suse.com>
+
+use strict;
+use warnings;
+use base 'y2_installbase';
+use testapi;
+use scheduler 'get_test_suite_data';
+
+sub run {
+    my $test_data = get_test_suite_data();
+    select_console 'root-console';
+    record_info("Check file", "Check if file /sys/firmware/efi/vars/SecureBoot-*/data exists");
+    assert_script_run("ls /sys/firmware/efi/vars/SecureBoot-*/data");
+    record_info("Check secure boot", "Check if secure boot option is set as expected");
+    my $is_secure = script_output("od -An -t u1 /sys/firmware/efi/vars/SecureBoot-*/data");
+    die "The secure boot option is not $test_data->{secure_boot}" if (($is_secure & ($test_data->{secure_boot} ne "enabled")) | (!($is_secure) & ($test_data->{secure_boot} ne "disabled")));
```

Comment:
Normally post-fixed notation is for short conditions, so it is more human-readable. I would suggest to write the condition as block here which would shorten the line as well.
As and alternative you could do the conversion before the assertion, something like:
`my $current_secure = script_output("od -An -t u1 /sys/firmware/efi/vars/SecureBoot-*/data") ? 'enabled' : 'disabled';
and then just use `assert_equals` of expected and actual.
By the way, in Perl the logical operator are `||` or `or` and `&&` or `and` and the binary operators are `&` and `|` which could lead to unexpected results.

Reply:
Actually i have the impression that it is wrong, as it will succeed even when the script_output fails

Reply:
> Actually i have the impression that it is wrong, as it will succeed even when the script_output fails

@b10n1k  let's try to organize the feedback to not confuse the person which PR is reviewed, it would be better if you quote my third paragraph to add your insight. As we agreed to reply to all comments, we need an action point for each of them, so I guess that might help.

Reply:
> make use of logical AND and not bitwise.
> With logical AND and OR you have higher precendence and you can avoid parenthesis, in contrast of && and ||.
> for example:
> record_info ("bla", "...") || die "fail!\n";
> versus
> record_info "bla", "..." || die "fail!\n";

same idea for this one, I think it is exactly the same point already mentioned, but some suggesiton

Reply:
> Actually i have the impression that it is wrong, as it will succeed even when the script_output fails

That is true, in the case of $test_data->{secure_boot}='enabled'. I can add 
assert_script_run("od -An -t u1 /sys/firmware/efi/vars/SecureBoot-*/data") 
before script_output. But why would we expect that script_output will fail since the file's sys/firmware/efi/vars/SecureBoot-*/data existence is already checked?

Reply:
it depends how you want to validate it I guess, so your module check this path that is only available for secure boot enabled (is like that?) but then you have a logic for both, enable or disabled. Maybe you can just check if non-secure 0 or secure 1 and then check for the path in case of enabled, make sense?

Reply:
Done